### PR TITLE
Implement `lr_scheduler`

### PR DESCRIPTION
**Pull request by tzuhanchang**
Implement `lr_scheduler` in HyPER.

### CODE REVIEW

**Comment by tzuhanchang** [comment]
Also need to add `lr_scheduler` related options in the configuration file.

**Comment by tzuhanchang** [comment]
> Also need to add `lr_scheduler` related options in the configuration file.

Ignore for now. If required, please open an issue [here](https://github.com/tzuhanchang/HyPER/issues).

**Comment by tzuhanchang** [comment]
This pull request implements the learning rate scheduler `torch.optim.lr_scheduler.ReduceLROnPlateau`; more could be added in future releases.
